test/schema.validation.test.js

```diff
@@ -1473,5 +1474,36 @@ describe('schema', function() {
         assert.ifError(err);
       });
     });
+
+    it('should validate required UUID fields correctly (gh-12991)', function() {
+      const uuidSchema = new mongoose.Schema({
+        _id: { type: mongoose.Schema.Types.UUID, required: true },
+        name: { type: mongoose.Schema.Types.String, required: true }
+      });
+
+      const uuidRefSchema = new mongoose.Schema({
+        _id: { type: mongoose.Schema.Types.UUID, required: true },
+        uuidRef: { type: mongoose.Schema.Types.UUID, ref: 'UUIDModel', required: true },
+        uuidNonRef: { type: mongoose.Schema.Types.UUID, required: true },
+        uuidRefNonRequired: { type: mongoose.Schema.Types.UUID, ref: 'UUIDModel' },
+        name: { type: mongoose.Schema.Types.String, required: true }
+      });
+
+      const UUIDModel = mongoose.model('UUIDModel', uuidSchema, 'uuids');
+
+      const UUIDRefModel = mongoose.model('UUIDRefModel', uuidRefSchema, 'uuidRefs');
+
+      const uuid = new UUIDModel({ _id: uuidv4(), name: 'uuidName' });
+      assert.ifError(uuid.validateSync());
+
+      const uuidRef = new UUIDRefModel({
+        _id: uuidv4(),
+        uuidRef: uuidv4(),
+        uuidNonRef: uuidv4(),
+        uuidRefNonRequired: uuidv4(),
+        name: 'uuidRefName'
+      });
+      assert.ifError(uuidRef.validateSync());
```

Comment:
Would be great to also test that validation fails if each of the `uuidRef`, `uuidNonRef`, etc. properties aren't set